places/src/db/db.rs

```diff
@@ -222,6 +223,11 @@ impl ConnectionUtil for PlacesDb {
 // ----------------------------- end of stuff that should be common --------------------
 
 fn define_functions(c: &Connection) -> Result<()> {
+    c.create_scalar_function("reverse_host", 1, true, move |ctx| {
+        let host = ctx.get::<String>(0)?;
+        let rev_host: String = host.graphemes(true).rev().flat_map(|g| g.chars().flat_map(|c| c.to_lowercase())).collect();
```

Comment:
I don't think we want to reverse by graphemes. This isn't stable across unicode versinos and regularly changes.

It's fairly slow, and given that storing the reversed host is an optimization, I'd be surprised if it were doing what we wanted).

Places reverses bytes: https://searchfox.org/mozilla-central/source/toolkit/components/places/SQLFunctions.cpp#942, Given that that's not really doable in rust without unsafe we should probably just reverse codepoints (e.g. `s.chars().rev().collect()`)

Does places reverse based on graphemes (? Are we sure we want to? This reversal isn't stable across unicode versions -- and in practice regularly changes.

Reply:
`Url::to_string()` is guaranteed to be ascii (punycoded host, percent-encoded everything else), so everything we use this on in practice should be also.

I did [some microbenchmarking](https://play.rust-lang.org/?gist=d20d7639a86cf1f814e09d3d14c032cb&version=nightly&mode=release&edition=2015) and we get a roughly 100x speedup if we operate on ascii chars versus grapheme clusters, (operating on codepoints is roughly the geometric mean of these, at 10x slower than ascii and 10x faster than grapheme clusters).